src/utils/periodic_utils.cpp

```diff
@@ -33,29 +33,68 @@
 namespace mrcpp {
 namespace periodic {
 
-template <int D> void indx_manipulation(NodeIndex<D> &idx, const std::array<bool, D> &periodic) {
-    if (idx.getScale() < 0) MSG_ABORT("Negative value in bit-shift");
-    std::array<int, D> translation;
-    int two_n = 1 << idx.getScale();
+template <int D> bool in_unit_cell(NodeIndex<D> idx) {
+    auto scale = idx.getScale();
+    if (scale < 0) MSG_ABORT("Negative value in bit-shift");
+    int two_n = 1 << (scale + 1);
+
+    int l[D];
     for (auto i = 0; i < D; i++) {
-        translation[i] = idx.getTranslation(i);
-        if (periodic[i]) {
-            if (translation[i] >= two_n) translation[i] = translation[i] % two_n;
-            if (translation[i] < 0) translation[i] = (translation[i] + 1) % two_n + two_n - 1;
+        l[i] = idx[i] + two_n / 2;
+        if (l[i] >= two_n) return false;
+        if (l[i] < 0) return false;
+    }
+
+    return true;
+}
+
+template <int D> void indx_manipulation(NodeIndex<D> &idx, const std::array<bool, D> &periodic) {
```

Comment:
Can you make it `index_manipulation`, no use in saving a single character in the name

Reply:
Fixed